awx_collection/plugins/modules/organization.py

```diff
@@ -159,49 +167,65 @@ def main():
         association_fields['instance_groups'] = []
         for item in instance_group_names:
             association_fields['instance_groups'].append(module.resolve_name_to_id('instance_groups', item))
+    elif instance_group_names is None and enforced_defaults:
+        association_fields['instance_groups'] = []
 
     notifications_start = module.params.get('notification_templates_started')
     if notifications_start is not None:
         association_fields['notification_templates_started'] = []
         for item in notifications_start:
             association_fields['notification_templates_started'].append(module.resolve_name_to_id('notification_templates', item))
+    elif notifications_start is None and enforced_defaults:
+        association_fields['notification_templates_started'] = []
```

Comment:
Instead of doing a bunch of `and enforced_defaults` could we just default the `association_fields` variable if enforced is true?
```
elif state == 'enforced':
    association_fields = {
         'instance_groups': []
         'notification_templates_started': []
         ...
    }
```

We could also do the same with org_fields.... or maybe even make this more generic with having a module call like:
```
   org_fields, association_fields = module.get_enforced_state('organizations')
 ```
 
You'd have to adjust the existing initialization of these fields but this might make a better pattern for all of the endpoints.

Reply:
Done in latest